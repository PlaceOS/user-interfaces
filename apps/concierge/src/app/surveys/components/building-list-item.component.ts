import { Component, OnInit, Input } from '@angular/core';
import { Building } from '@placeos/organisation';
import { Router, ActivatedRoute } from '@angular/router';
import { BuildingListItemService } from '../services/building-list-item.service';
import { shareReplay } from 'rxjs/operators';
import { SettingsService } from '@placeos/common';

@Component({
    selector: 'building-list-item',
    template: `
        <section
            class="building-wrapper bg-base-100 my-4 mx-auto rounded-md overflow-hidden"
        >
            <div *ngIf="building.image" class="image-container">
                <img
                    auth
                    class="flex object-fill "
                    [source]="building.image"
                    alt="image of building"
                />
            </div>
            <div *ngIf="!building.image" class="image-container">
                <span> <app-icon class="domain-icon">domain</app-icon></span>
            </div>
            <div class="details-container my-4 mx-8 w-full relative">
                <div class="location-wrapper">
                    <app-icon
                        class="location-icon"
                        aria-hidden="false"
                        aria-label="Material icon of location pointer"
                        >location_on</app-icon
                    >
                    <span> {{ building.address }} </span>
                </div>

                <span class="building-title">
                    {{ building.display_name }}
                </span>
                <ng-container
                    *ngIf="!(loading$ | async)?.length; else loadState"
                >
                    <ng-container *ngIf="stats$ | async as stats">
                        <div class="flex py-4 justify-end space-x-2">
                            <div class="flex flex-col items-center flex-1">
                                <h3>Live Surveys</h3>
                                <p class="text-4xl">
                                    {{ stats?.lives || 0 }}
                                </p>
                            </div>
                            <div class="flex flex-col items-center flex-1">
                                <h3>Draft Surveys</h3>
                                <p class="text-4xl">
                                    {{ stats?.drafts || 0 }}
                                </p>
                            </div>
                            <div class="flex flex-col items-center flex-1">
                                <h3>Total Answers</h3>
                                <p class="text-4xl">
                                    {{ stats?.responses || 0 }}
                                </p>
                            </div>
                        </div>
                    </ng-container>
                </ng-container>
                <div class="flex justify-end w-full mt-4">
                    <button btn matRipple class="inverse" (click)="navigate()">
                        <span class="ml-2">View</span>
                        <app-icon class="ml-1 text-2xl">chevron_right</app-icon>
                    </button>
                </div>
            </div>
        </section>
        <ng-template #loadState>
            <div class="flex absolute inset-0 opacity-60 bg-base-100 z-10">
                <div class="flex flex-col m-auto items-center">
                    <mat-spinner [diameter]="32"></mat-spinner>
                    <span>{{ loading$ | async }}</span>
                </div>
            </div>
        </ng-template>
    `,
    styles: [
        `
            .building-wrapper {
                display: flex;
                position: relative;
                flex-direction: row;
                position: relative;
                justify-content: flex-start;
                height: 260px;
                max-width: 68rem;
                border: 1px solid #e6e6e6;
                box-shadow: 0px 2px 4px rgba(5, 28, 44, 0.1);
            }
            .options {
                display: flex;
                align-items: center;
                justify-content: center;
                background-color: var(--b1);
                position: absolute;
                height: 48px;
                width: 48px;
                top: 10px;
                left: 10px;
                border: 1px solid #e6e6e6;
                border-radius: 3px;
            }
            .menu-wrapper {
                display: flex;
                flex-direction: column;
                height: 90px;
                width: 135px;
                justify-content: center;
            }
            .image-container {
                display: flex;
                height: 100%;
                width: 380px;
                background-color: rgba(0, 0, 0, 0.22);
                align-items: center;
                justify-content: center;
            }
            .details-container {
                display: flex;
                flex-direction: column;
                min-width: 600px;
            }
            .location-wrapper {
                display: flex;
                flex-direction: row;
                align-items: center;
                margin-bottom: 20px;
            }
            .location-icon {
                color: #5295f7;
                margin-right: 10px;
            }
            .domain-icon {
                display: flex;
                color: #fff;
                height: 100%;
                width: 100%;
            }
            .building-title {
                font-size: 18px;
                font-weight: 500;
            }
            .details-text {
                display: flex;
                flex-direction: column;
                font-size: 14px;
                font-weight: 400;
                color: #020202;
                margin-bottom: 10px;
                justify-content: space-between;
                height: 70px;
            }
            .button-container {
                display: flex;
                position: absolute;
                bottom: 20px;
                right: 20px;
            }
            /* .view-button {
                display: inline-flex;
                color: #292f5b;
                background-color: var(--b1);
                border-radius: 2px;
                margin: 20px 0px 20px 20px;
                justify-content: center;
                padding-left: 20px;
            }
            .view-button span {
                display: inline-flex;
            }
            .view-button app-icon {
                display: flex;
                justify-content: center;
                align-items: center;
            } */
        `,
    ],
    providers: [BuildingListItemService],
})
export class BuildingListItemComponent implements OnInit {
    @Input() building: Building | any;

    loading$ = this.service.loading$.pipe(shareReplay(1));
    stats$ = this.service.stats$.pipe(shareReplay(1));

    constructor(
        private _settings: SettingsService,
        private router: Router,
        private service: BuildingListItemService
    ) {}

    ngOnInit(): void {
        this.service.initStats(this.building.id);
    }

    navigate(): void {
        this.router.navigate([
            this._settings.get('app.default_route').includes('new')
                ? '/surveys/new'
                : '/survey',
            'survey-list',
            this.building.id,
        ]);
    }
}
