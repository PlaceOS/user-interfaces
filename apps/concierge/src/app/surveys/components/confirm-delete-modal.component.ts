import { Component, OnInit, Inject } from '@angular/core';
import { MatDialog, MatDialogRef } from '@angular/material/dialog';

@Component({
    selector: 'confirm-delete-modal',
    template: `
        <aside>
            <span
                >Are you sure you want to permanently delete this survey?</span
            >
            <div class="button-wrapper">
                <button
                    mat-button
                    (click)="cancel()"
                    class="cancel-button"
                    aria-label="do not delete survey"
                >
                    No
                </button>
                <button
                    mat-flat-button
                    (click)="delete()"
                    class="delete-button"
                    aria-label="delete survey"
                >
                    Yes Delete
                </button>
            </div>
        </aside>
    `,
    styles: [
        `
            aside {
                margin: 20px;
            }
            .button-wrapper {
                display: flex;
                padding-top: 20px;
                width: 100%;
                justify-content: space-between;
            }
            .cancel-button {
                background-color: #fff;
                border: 1px solid rgba(0, 0, 0, 0.12);
                color: black;
                border-radius: 2px;
                font-weight: 700;
                padding: 6px 20px;
            }
            .delete-button {
                background-color: #e12e40;
                color: #fff;
                font-weight: 700;
                border-radius: 2px;
                padding: 6px 20px;
            }
        `,
    ],
})
export class ConfirmDeleteModalComponent implements OnInit {
    constructor(public dialogRef: MatDialogRef<ConfirmDeleteModalComponent>) {}

    ngOnInit(): void {}

    cancel(): void {
        this.dialogRef.close(false);
    }

    delete() {
        this.dialogRef.close(true);
    }
}
