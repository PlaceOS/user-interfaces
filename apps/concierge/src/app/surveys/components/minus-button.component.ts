import { Component, OnInit } from '@angular/core';

@Component({
    selector: 'minus-button',
    template: `
        <div class="button-container">
            <button mat-mini-fab aria-label="Button with plus icon">
                <mat-icon>remove</mat-icon>
            </button>
        </div>
    `,
    styles: [
        `
            .button-container {
                display: flex;
                justify-content: center;
                width: 40px;
            }

            .button-container button {
                display: flex;
                position: relative;
                background-color: #9a2d2d;
                height: 20px;
                width: 20px;
                justify-content: center;
            }

            .button-container button mat-icon {
                position: absolute;
                top: 0;
                left: 0;
                margin-left: -2px;
                margin-top: 1px;
                font-size: 18px;
                font-weight: 700;
            }
        `,
    ],
})
export class MinusButtonComponent implements OnInit {
    constructor() {}

    ngOnInit(): void {}
}
