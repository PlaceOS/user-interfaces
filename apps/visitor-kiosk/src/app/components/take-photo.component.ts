import {
    Component,
    OnInit,
    HostBinding,
    OnDestroy,
    Output,
    EventEmitter,
} from '@angular/core';

@Component({
    selector: 'a-take-photo',
    template: `
        <div
            name="camera"
            class="flex flex-col items-center justify-center rounded overflow-hidden relative"
        >
            <mat-spinner diameter="32"></mat-spinner>
            <div class="text">Please wait...</div>
            <video
                id="video"
                #video
                autoplay
                width="360"
                height="400"
                class="absolute inset-0"
            ></video>
            <canvas
                id="canvas"
                #canvas
                class="absolute inset-0"
                width="360"
                height="400"
            ></canvas>
        </div>
        <div class="flex items-center justify-center mt-4 space-x-2">
            <button
                class="take-photo"
                *ngIf="!hasPhoto; else accept_state"
                mat-button
                (click)="takePhoto()"
            >
                Take Photo
            </button>
        </div>
        <ng-template #accept_state>
            <button class="inverse" mat-button (click)="cancelPhoto()">
                Cancel
            </button>
            <button mat-button (click)="acceptPhoto()">Accept</button>
        </ng-template>
    `,
    styles: [
        `
            :host {
                display: flex;
                flex-direction: column;
                align-items: center;
            }

            [name='camera'] {
                width: 22.5rem;
                height: 25rem;
            }

            button {
                width: 8rem;
            }
        `,
    ],
})
export class TakePhotoComponent implements OnInit, OnDestroy {
    @Output() public photoAccepted = new EventEmitter();
    @HostBinding('class.has-photo') public hasPhoto = false;
    public video: HTMLVideoElement;
    public canvas: HTMLCanvasElement;
    private canvasContext = null;

    private constraints = {
        audio: false,
        video: {
            width: { min: 360, max: 360 },
            height: { min: 400, max: 400 },
        },
    };

    public imgSrc = null;

    public ngOnInit() {
        setTimeout(() => this.getCanvasContext(), 500);
    }

    public ngOnDestroy() {
        this.stopCapture();
        this.video = null;
        this.canvasContext = null;
        this.canvas = null;
        this.imgSrc = null;
    }

    private startCapture() {
        this.imgSrc = null;
        console.log('TakePhotoComponent::startCapture');
        this.video = document.getElementById('video') as any;
        if (!this.video) {
            return setTimeout(() => this.startCapture(), 200);
        } else {
            navigator.mediaDevices
                .getUserMedia(this.constraints)
                .then((stream) => {
                    this.video.srcObject = stream;
                });
        }
    }

    private stopCapture() {
        console.log('TakePhotoComponent::stopCapture');
        if (this.video?.srcObject) {
            (this.video.srcObject as any)
                .getVideoTracks()
                .forEach((track) => track.stop());
        }
    }

    private getCanvasContext() {
        console.log('TakePhotoComponent::getCanvasContext');
        this.canvas = document.getElementById('canvas') as any;
        if (!this.canvas) {
            return setTimeout(() => this.getCanvasContext(), 200);
        }
        this.canvasContext = this.canvas.getContext('2d');
        setTimeout(() => this.startCapture(), 200);
    }

    public takePhoto() {
        console.log('TakePhotoComponent::takePhoto');
        this.canvasContext.drawImage(
            this.video,
            0,
            0,
            this.canvas.width,
            this.canvas.height
        );
        this.hasPhoto = true;
        this.stopCapture();
    }

    public cancelPhoto() {
        console.log('TakePhotoComponent::cancelPhoto');
        this.canvasContext.clearRect(
            0,
            0,
            this.canvas.width,
            this.canvas.height
        );
        this.hasPhoto = false;
        this.startCapture();
    }

    public acceptPhoto() {
        try {
            this.imgSrc = this.canvas.toDataURL('image/jpeg', 0.75);
            this.photoAccepted.emit(this.imgSrc);
            console.log('TakePhotoComponent::acceptPhoto Success');
        } catch (err) {
            console.error(
                'TakePhotoComponent::acceptPhoto Error converting image',
                err
            );
            this.photoAccepted.emit(null);
            this.cancelPhoto();
        }
    }
}
