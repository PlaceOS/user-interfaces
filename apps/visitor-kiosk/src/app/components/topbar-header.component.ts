import { Component } from '@angular/core';
import { format } from 'date-fns';

@Component({
    selector: 'a-topbar-header',
    template: `
        <div class="flex items-center justify-between bg-secondary shadow">
            <a
                mat-button
                [routerLink]="['/']"
                class="h-full flex flex-col justify-center px-4"
            >
                <img class="h-10 my-2" src="assets/logo-dark.svg" />
            </a>
            <div
                class="ml-auto h-full flex flex-col justify-center text-white px-4"
            >
                {{ time }}
            </div>
        </div>
    `,
    styles: [
        `
            :host > div {
                height: 3.5rem;
            }

            a {
                border-radius: 0;
            }

            img {
                max-height: 2.5rem;
                max-width: 50vw;
            }
        `,
    ],
})
export class TopbarHeaderComponent {
    public date: number;

    public get time() {
        return format(this.date || Date.now(), 'EEEE, MMMM do, h:mma');
    }
}
