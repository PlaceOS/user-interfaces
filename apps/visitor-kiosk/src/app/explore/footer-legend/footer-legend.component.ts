import { Component } from '@angular/core';

import { Identity } from '@user-interfaces/common';

@Component({
  selector: 'a-footer-legend',
  templateUrl: './footer-legend.component.html',
  styleUrls: ['./footer-legend.component.scss']
})
export class FooterLegendComponent {
    /** Whether icon legend should be shown */
    public show: boolean;
    /** List of colour legend items */
    public readonly colours: readonly Identity[] = [
        { id: 'in-use', name: 'In Use', colour: '#C52A1A', dark: true },
        { id: 'available', name: 'Available', colour: '#22992E', dark: true },
        { id: 'not-bookable', name: 'Not Bookable', colour: '#7D7D7D', dark: true },
        { id: 'collab', name: 'Collaborative', colour: '#FED769' },
        { id: 'focus', name: 'Focus', colour: '#F5A478' }
    ];
    /** List of icon legend items */
    public readonly icon_list: readonly Identity[] = [
        { id: '', name: 'Lifts Up/Down', icon: { type: 'img', src: 'assets/img/lift.svg' } },
        { id: '', name: 'Fire Extinguisher', icon: { type: 'img', src: 'assets/img/fireExt.svg' } },
        { id: '', name: 'First Aid & AED', icon: { type: 'img', src: 'assets/img/aid.svg' } },
        { id: '', name: 'Cafe/Service', icon: { type: 'img', src: 'assets/img/Cafe.svg' } },
        { id: '', name: 'Mail/Noticeboard', icon: { type: 'img', src: 'assets/img/mail.svg' } },
        { id: '', name: 'Restrooms M/F', icon: { type: 'img', src: 'assets/img/Restrooms.svg' } },
        { id: '', name: 'Map Kiosk', icon: { type: 'img', src: 'assets/img/kiosk.svg' } },
        { id: '', name: 'Screen', icon: { type: 'img', src: 'assets/img/Screen.svg' } },
        { id: '', name: 'Projector', icon: { type: 'img', src: 'assets/img/Projector.svg' } },
        { id: '', name: 'Showers', icon: { type: 'img', src: 'assets/img/shower.svg' } },
        { id: '', name: 'Chairs', icon: { type: 'img', src: 'assets/img/chair.svg' } },
        { id: '', name: 'Chairs', icon: { type: 'img', src: 'assets/img/chair2.svg' } },
        { id: '', name: 'Chairs/Couch', icon: { type: 'img', src: 'assets/img/couch.svg' } },
        { id: '', name: 'Chairs/Couch', icon: { type: 'img', src: 'assets/img/couch2.svg' } },
        { id: '', name: 'Tables', icon: { type: 'img', src: 'assets/img/Table1.svg' } },
        { id: '', name: 'Tables', icon: { type: 'img', src: 'assets/img/Table2.svg' } },
        { id: '', name: 'Tables', icon: { type: 'img', src: 'assets/img/Table3.svg' } },
        { id: '', name: 'Tables', icon: { type: 'img', src: 'assets/img/Table4.svg' } },
        { id: '', name: 'Tables', icon: { type: 'img', src: 'assets/img/Table5.svg' } },
        { id: '', name: 'Tables', icon: { type: 'img', src: 'assets/img/Table6.svg' } },
        { id: '', name: 'Tables', icon: { type: 'img', src: 'assets/img/Table7.svg' } },
        { id: '', name: 'Funzone Games', icon: { type: 'img', src: 'assets/img/Tennis.svg' } },
        { id: '', name: 'Concierge Pods', icon: { type: 'img', src: 'assets/img/Pod.svg' } },
        { id: '', name: 'Phone Booths', icon: { type: 'img', src: 'assets/img/booth.svg' } }
    ];
}
