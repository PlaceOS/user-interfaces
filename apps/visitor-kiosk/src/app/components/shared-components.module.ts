import { NgModule, Type } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { ScrollingModule } from '@angular/cdk/scrolling';
import { EditorModule } from '@tinymce/tinymce-angular';

import { MatButtonModule } from '@angular/material/button';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatDialogModule } from '@angular/material/dialog';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatInputModule } from '@angular/material/input';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatProgressBarModule } from '@angular/material/progress-bar';
import { MatAutocompleteModule } from '@angular/material/autocomplete';
import { MatSelectModule } from '@angular/material/select';
import { MatChipsModule } from '@angular/material/chips';
import { MatRippleModule } from '@angular/material/core';
import { MatMenuModule } from '@angular/material/menu';
import { MatBottomSheetModule } from '@angular/material/bottom-sheet';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatRadioModule } from '@angular/material/radio';

import { ComponentsModule } from '@placeos/components';

import { TopbarHeaderComponent } from './topbar-header.component';
import { TakePhotoComponent } from './take-photo.component';

const COMPONENTS: Type<any>[] = [TopbarHeaderComponent, TakePhotoComponent];

const MATERIAL_MODULES: any[] = [
    MatButtonModule,
    MatSnackBarModule,
    MatDialogModule,
    MatFormFieldModule,
    MatInputModule,
    MatProgressSpinnerModule,
    MatProgressBarModule,
    MatAutocompleteModule,
    MatSelectModule,
    MatChipsModule,
    MatRippleModule,
    MatMenuModule,
    MatBottomSheetModule,
    MatCheckboxModule,
    MatRadioModule,
];

const CDK_MODULES: any[] = [ScrollingModule];

@NgModule({
    declarations: [...COMPONENTS],
    imports: [
        CommonModule,
        FormsModule,
        ReactiveFormsModule,
        ComponentsModule,
        RouterModule,
        EditorModule,
        ...MATERIAL_MODULES,
        ...CDK_MODULES,
    ],
    exports: [
        ComponentsModule,
        ...COMPONENTS,
        ...MATERIAL_MODULES,
        ...CDK_MODULES,
    ],
})
export class SharedComponentsModule {}
