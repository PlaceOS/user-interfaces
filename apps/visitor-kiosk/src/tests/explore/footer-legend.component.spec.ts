import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { IconComponent } from '@placeos/components';
import { MockComponent } from 'ng-mocks';
import { FooterLegendComponent } from '../../app/explore/footer-legend.component';

describe('FooterLegendComponent', () => {
    let spectator: Spectator<FooterLegendComponent>;
    const createComponent = createComponentFactory({
        component: FooterLegendComponent,
        declarations: [MockComponent(IconComponent)],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });
});
