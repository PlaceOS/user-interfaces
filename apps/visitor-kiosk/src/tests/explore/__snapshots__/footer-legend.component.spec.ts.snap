// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLegendComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="relative bg-white shadow w-full"
  >
    <div
      class="absolute top-0 left-1/2 transform -translate-y-full -translate-x-1/2 px-2 pt-2 overflow-hidden"
    >
      <button
        class="flex items-center justify-center bg-white shadow rounded-t w-32 h-8 border-none text-black"
        matripple=""
      >
        <app-icon>
          keyboard_arrow_up
        </app-icon>
      </button>
    </div>
    <div
      class="flex items-center px-2 py-1 text-xl"
    >
      <div
        class="flex-1"
      >
        Keys
      </div>
    </div>
    <div
      class="flex items-center justify-center p-2 border border-gray-100 space-x-2"
    >
      <div
        class="colour px-4 py-2 rounded shadow flex-1 text-center dark"
        style="background-color: rgb(197, 42, 26);"
      >
         In Use 
      </div>
      <div
        class="colour px-4 py-2 rounded shadow flex-1 text-center dark"
        style="background-color: rgb(34, 153, 46);"
      >
         Available 
      </div>
      <div
        class="colour px-4 py-2 rounded shadow flex-1 text-center dark"
        style="background-color: rgb(125, 125, 125);"
      >
         Not Bookable 
      </div>
      <div
        class="colour px-4 py-2 rounded shadow flex-1 text-center"
        style="background-color: rgb(254, 215, 105);"
      >
         Collaborative 
      </div>
      <div
        class="colour px-4 py-2 rounded shadow flex-1 text-center"
        style="background-color: rgb(245, 164, 120);"
      >
         Focus 
      </div>
      
    </div>
    <div
      class="flex justify-center flex-wrap h-0 w-full border border-gray-100 overflow-auto max-h-[25vh]"
      icon-legend=""
    >
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Lifts Up/Down 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Fire Extinguisher 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           First Aid & AED 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Cafe/Service 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Mail/Noticeboard 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Restrooms M/F 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Map Kiosk 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Screen 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Projector 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Showers 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Chairs 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Chairs 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Chairs/Couch 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Chairs/Couch 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Tables 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Tables 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Tables 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Tables 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Tables 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Tables 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Tables 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Funzone Games 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Concierge Pods 
        </div>
      </div>
      <div
        class="flex flex-col items-center justify-center h-[12.4vw] w-[12.4vw] text-[2vw] border border-gray-100 flex-grow"
      >
        <app-icon
          class="text-[3em]"
        />
        <div
          class="text-[0.625em] font-normal"
        >
           Phone Booths 
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
